src/espina/client/router.js

```diff
@@ -0,0 +1,101 @@
+/*global window NODE_ENV*/
+import {stringify} from 'query-string';
+import BaseRouter from 'espina/shared/router'
+import Scroll from "scroll-js" 
+
+export default class Router extends BaseRouter {
+
+  get locale(){
+    return Router.locale();
+  }
+
+  initializeHistory(createHistory, store,fnChange) {
+    let router = this;
+    router.history = createHistory();
+    router.history.listen(router.onLocationChange.bind(router, store));
+    if (fnChange) router.history.listen(fnChange);
+  }
+
+  // Changing Route
+
+  // this will cause onLocationChange to fire with
+  // the new location.
+  pushRoute(route_name, action, payload,params,hash){
+    let router = this,
+        route = router.routes.findByName(route_name);
+
+    action = {
+      type: action ? action.getType() : 'UPDATE_LOCATION',
+      payload: payload,
+      no_scroll: payload ? payload.no_scroll : false,
+      transition:true
+    };
+
```

Comment:
I will circle around and deprecate the implicit support for `redux-act`. I think users of the repo will be able to just pass in `action.getType()` so we don't have to call it here.